addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4905,6 +4902,27 @@ registry.ImageOptimize = ImageHandlerOption.extend({
             $(btn).prepend(image);
         });
     },
+    /**
+     * Gets the CSS value of a color variable name so it can be used on shapes.
+     *
+     * @param {string} color
+     * @returns {string}
+     */
+    _getCSSColorValue(color) {
+        if (ColorpickerWidget.isCSSColor(color)) {
+            return color;
+        }
+        const doc = this.options.document;
+        if (doc && doc.querySelector('.o_mass_mailing_iframe')) {
+            const tempEl = doc.body.appendChild(doc.createElement('div'));
+            tempEl.className = `bg-${color}`;
+            const colorValue = window.getComputedStyle(tempEl).getPropertyValue("background-color").trim();
+            tempEl.parentNode.removeChild(tempEl);
+            return ColorpickerWidget.normalizeCSSColor(colorValue).replace(/"/g, "'");
+        } else {
+            return weUtils.getCSSVariableValue(color);
+        }
+    },
```

Comment:
You mention a mass mailing class inside of the web_editor app -> not ok. But this is going in the right direction: what you have to do is define that `_getCSSColorValue` here with only the normal "website" case -> reading the CSS variable. Then, in mass mailing (something mass_mailing_editor.js), you can **include** this JS class to change the `_getCSSColorValue` method behavior.